Sources/DistributedActors/Cluster/Downing/DowningStrategy.swift

```diff
@@ -33,88 +32,93 @@ public protocol DowningStrategy {
+        ps._wellKnown = true
+        return ps
     }
 
-    var behavior: _Behavior<Message> {
-        .setup { context in
-            let clusterEventSubRef = context.subReceive(Cluster.Event.self) { event in
-                do {
-                    try self.receiveClusterEvent(context, event: event)
-                } catch {
-                    context.log.warning("Error while handling cluster event: [\(error)]\(type(of: error))")
-                }
-            }
-            context.system.cluster.events.subscribe(clusterEventSubRef)
+    let strategy: DowningStrategy
 
-            return .receiveMessage { message in
-                switch message {
-                case .timeout(let member):
-                    let directive = self.strategy.onTimeout(member)
-                    context.log.debug("Received timeout for [\(member)], resulting in: \(directive)")
-                    self.interpret(context, directive)
-                }
+    /// `Task` for subscribing to cluster events.
+    private var eventsListeningTask: Task<Void, Error>?
 
-                return .same
+    private lazy var timers = ActorTimers<DowningStrategyShell>(self)
+
+    init(_ strategy: DowningStrategy, system: ActorSystem) async {
+        self.strategy = strategy
+        self.actorSystem = system
+        self.eventsListeningTask = Task {
+            for await event in await system.cluster.events {
+                try self.receiveClusterEvent(event)
             }
         }
     }
 
-    func receiveClusterEvent(_ context: _ActorContext<Message>, event: Cluster.Event) throws {
+    deinit {
+        self.eventsListeningTask?.cancel()
+    }
+
+    func receiveClusterEvent(_ event: Cluster.Event) throws {
         let directive: DowningStrategyDirective = try self.strategy.onClusterEvent(event: event)
-        self.interpret(context, directive)
+        self.interpret(directive: directive)
     }
 
-    func interpret(_ context: _ActorContext<Message>, _ directive: DowningStrategyDirective) {
+    func interpret(directive: DowningStrategyDirective) {
         switch directive {
         case .markAsDown(let members):
-            self.markAsDown(context, members: members)
+            self.markAsDown(members: members)
 
-        case .startTimer(let key, let message, let delay):
-            context.log.trace("Start timer \(key), message: \(message), delay: \(delay)")
-            context.timers.startSingle(key: key, message: message, delay: delay)
+        case .startTimer(let key, let member, let delay):
+            self.actorSystem.log.trace("Start timer \(key), member: \(member), delay: \(delay)")
+            self.timers.startSingle(key: key, delay: delay) {
+                self.onTimeout(member: member)
+            }
         case .cancelTimer(let key):
-            context.log.trace("Cancel timer \(key)")
-            context.timers.cancel(for: key)
+            self.actorSystem.log.trace("Cancel timer \(key)")
```

Comment:
You can create a logger instance inside the actor body like: lazy var logger = Logger(actor: self) this will include its ID in logging then :)

Reply:
amended https://github.com/apple/swift-distributed-actors/pull/938/commits/4dad9a4236115e4bcf82cf9806962b99ef39777d